### PR TITLE
cc2538/include/periph_cpu.h: Changed UART configuration format for cc2538.

### PR DESCRIPTION
**Pull request by danpetry**
### Contribution description

Changed the style of the UART configuration for different boards, from
a define based configuration to one based on an array of structs, one
struct for each UART, with the format of the struct defined in
cc2538/include/periph_cpu.h.

  - Defined the fields of the struct in periph_cpu.h
  - Removed the compilation includes that were in uart.c for each UART
  - Implemented a generic ISR subroutine for clarity
  - Implemented the configuration on openmote-cc2538 and cc2538dk boards.

Notes to reviewer:

  - Openmote has been tested, but cc2538dk has not, due to lack of the
    latter board.
  - Some information about UART1 on the cc2538dk has been deleted, because
    UART1 was not being used. This information would have gone in the
    second index of the config array - but presence of this second index
    would have enabled UART1, in the new style of configuration. The
    same applies for the flow control for UART0 on the cc2538dk; flow
    control is only available on UART1 for the cc2538 cpu.
  - The switch on the UART number has been deleted. The default statement
    offered protection on the "uart" variable being outside the range allowed
    by the cpu. This protection is no longer present. What effect might this
    have? Suggest that the uart_t type should be enumerated on a per-cpu
    basis, to restrict the range, but would this be worth the effort to change?

### Issues/PRs references

Ticks the UART checkbox for cc2538 in https://github.com/RIOT-OS/RIOT/issues/7941.

### CODE REVIEW

File: boards/cc2538dk/include/periph_conf.h [hunk 1]
```diff
@@ -58,34 +58,31 @@ static const timer_conf_t timer_config[] = {
 #define TIMER_IRQ_PRIO      1
 /** @} */
 
+
```
Comment (aabadie) [inline boards/cc2538dk/include/periph_conf.h]: this blankline could be removed

File: boards/cc2538dk/include/periph_conf.h [hunk 2]
```diff
@@ -58,34 +58,31 @@ static const timer_conf_t timer_config[] = {
...
+ * @name    UART configuration
  * @{
  */
-#define UART_NUMOF          1
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
+        .dev      = (cc2538_uart_t *)0x4000c000,
+        .rx_pin   = GPIO_PIN(0,0),
+        .tx_pin   = GPIO_PIN(0,1),
+	.irq      = UART0_IRQn,
```
Comment (aabadie) [inline boards/cc2538dk/include/periph_conf.h]: vertical alignment shift

File: boards/openmote-cc2538/include/periph_conf.h [hunk 3]
```diff
@@ -86,17 +86,26 @@ static const adc_conf_t adc_config[] = {
...
  */
-#define UART_NUMOF          (1U)
-#define UART_0_EN           1
-#define UART_IRQ_PRIO       1
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
+        .dev      = (cc2538_uart_t *)0x4000c000,
+        .rx_pin   = GPIO_PIN(0,0),
+        .tx_pin   = GPIO_PIN(0,1),
+	.irq      = UART0_IRQn,
```
Comment (aabadie) [inline boards/openmote-cc2538/include/periph_conf.h]: alignment

File: cpu/cc2538/periph/uart.c [hunk 4]
```diff
@@ -106,52 +103,10 @@ static void reset(cc2538_uart_t *u)
...
-        reset(UART_1_DEV);
-    }
-
-    cortexm_isr_end();
-}
-#endif /* UART_1_EN */
 
 static int init_base(uart_t uart, uint32_t baudrate);
 
+
```
Comment (aabadie) [inline cpu/cc2538/periph/uart.c]: not needed

File: cpu/cc2538/periph/uart.c [hunk 5]
```diff
@@ -224,20 +147,19 @@ static int init_base(uart_t uart, uint32_t baudrate)
     u->CC = 0;
 
     /* On the CC2538, hardware flow control is supported only on UART1 */
-    if (u == UART1) {
-#ifdef UART_1_RTS_PIN
-        gpio_init_af(UART_1_RTS_PIN, UART1_RTS, GPIO_OUT);
+//    if (u == UART1) {
```
Comment (aabadie) [inline cpu/cc2538/periph/uart.c]: please remove this comment

File: cpu/cc2538/periph/uart.c [hunk 6]
```diff
@@ -277,27 +199,14 @@ static int init_base(uart_t uart, uint32_t baudrate)
     u->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 
     return UART_OK;
-#endif /* UART_0_EN || UART_1_EN */
+
```
Comment (aabadie) [inline cpu/cc2538/periph/uart.c]: not needed

**Comment by aabadie** [comment]
I would also split the commits in 3: 1 for the change in cpu and one for each boards config update.

**Comment by danpetry** [comment]
Ok I've done these changes. Will check the CI report shortly.

**Comment by danpetry** [comment]
@smlng - would you be able to test these changes on the cc2538dk at all please? I don't have one here. :)

**Comment by smlng** [comment]
@danpetry, maybe @A-Paul can help, he was working with that dev board.

**Comment by danpetry** [comment]
Travis is passing, ok to set the murdock flag?

**Comment by aabadie** [comment]
Maybe reword the commit messages:
* 0b1f202: cpu/cc2538: Changed configuration interface of cc2538 uart ...
* prepend the others with `boards/`: 'boards/cc2538dk: ...' and 'boards/openmote-cc2538: ...'

**Comment by danpetry** [comment]
Updated and rebased

**Comment by aabadie** [comment]
2 other boards are also using the CC2538 and needs update: `remote-reva` and `remote-revb`

**Comment by danpetry** [comment]
also `remote-pa`

**Comment by danpetry** [comment]
Hi Chaps

I've only got one of the boards with this CPU on it. Could anybody please help with testing? All you need to do is build a binary, flash it, and make sure that you can communicate with the firmware over UART0. None of the boards are using UARTs 1 and above.

- [ ] Openmote-cc2538

- [ ] cc2538dk

- [ ] remote-pa

- [ ] remote-reva

- [ ] remote-revb

Hauke tells me that @smlng @PeterKietzmann @cgundogan - you might have some boards around your way - would you kindly be able to assist?

**Comment by smlng** [comment]
I got remove-rev sitting on my desks (will test today). @PeterKietzmann should have the `reva` and `pa`, don't you? And as said @A-Paul worked with `cc2538dk`.

File: boards/cc2538dk/include/periph_conf.h [hunk 7]
```diff
@@ -59,33 +59,29 @@ static const timer_conf_t timer_config[] = {
...
 
 /**
- * @name UART configuration
+ * @name    UART configuration
  * @{
  */
-#define UART_NUMOF          1
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
+        .dev      = (cc2538_uart_t *)0x4000c000,
```
Comment (smlng) [inline boards/cc2538dk/include/periph_conf.h]: I'd rather not have such address directly in the config, there should be defines for that. Or use numbers to index the UART used (i.e., `0`, `1`) and have a helper function in the driver to retrieve the corresponding `cc2538_uart_t`. See the timer implementation for instance, which has `static inline cc2538_gptimer_t *dev(tim_t Tim)` function.

File: boards/cc2538dk/include/periph_conf.h [hunk 8]
```diff
@@ -59,33 +59,29 @@ static const timer_conf_t timer_config[] = {
...
+        .rx_pin   = GPIO_PIN(0,0),
```
Comment (smlng) [inline boards/cc2538dk/include/periph_conf.h]: please add space, i.e. `(0, 0)`, here and below

**Comment by smlng** [comment]
Just  tested with remote-revb: works!

File: boards/cc2538dk/include/periph_conf.h (hunk 7 above)
Comment (danpetry) [reply boards/cc2538dk/include/periph_conf.h]: ok cool - do you have any advice about consistency - i.e. which of those two approaches is more widely implemented? In the sam0common drivers (IIRC) there's an implementation that's half way between, as well, where there's a helper function to retrieve the appropriate entry in the config array, but this doesn't solve the above problem.

Comment (haukepetersen) [reply boards/cc2538dk/include/periph_conf.h]: IMHO this is not about which approach is implemented more often, but a matter of efficiency: using a define with the UART base addresses here is more efficient than using an integer index and decoding it to the base address in the driver's c-file... So I see no reason for using an integer index here.

File: boards/cc2538dk/include/periph_conf.h [hunk 9]
```diff
@@ -59,33 +59,29 @@ static const timer_conf_t timer_config[] = {
 /** @} */
 
 /**
- * @name UART configuration
+ * @name    UART configuration
  * @{
  */
-#define UART_NUMOF          1
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
```
Comment (haukepetersen) [inline boards/cc2538dk/include/periph_conf.h]: What does `/* Virtual COM Port */` mean here? Configuring an UART is neither virtual, nor is it a (Windows) COM port in this context :-)

File: boards/cc2538dk/include/periph_conf.h [hunk 10]
```diff
@@ -59,33 +59,29 @@ static const timer_conf_t timer_config[] = {
...
 
-#define UART_0_EN           1
-#define UART_1_EN           0
-#define UART_2_EN           0
-#define UART_3_EN           0
+/* interrupt function name mapping */
+#define UART_0_ISR          isr_uart0
 
+/* macros common across all UARTs */
 #define UART_IRQ_PRIO       1
```
Comment (haukepetersen) [inline boards/cc2538dk/include/periph_conf.h]: no need for this value here: all IRQs are set to prio 1 per default in the cortexm initialization.

File: cpu/cc2538/include/periph_cpu.h [hunk 11]
```diff
@@ -112,6 +112,23 @@ typedef enum {
...
+/**
+ * @name   UART device configuration
+ * @{
+ */
+typedef struct {
+    cc2538_uart_t *dev;       /**< pointer to the used UART device */
+    gpio_t rx_pin;            /**< pin used for RX */
+    gpio_t tx_pin;            /**< pin used for TX */
+    IRQn_Type irq;            /**< interrupt number of the UART's multifunction interrupt */
+    cc2538_ioc_pin_t rxd;     /**< function assigned to the RX pin */
```
Comment (haukepetersen) [inline cpu/cc2538/include/periph_cpu.h]: why make `rxd` and `txd` configurable by the board? These values are static for the RX and TX pins, so they can be hard-coded in the uart driver implementation, right?!

File: cpu/cc2538/include/periph_cpu.h [hunk 12]
```diff
@@ -112,6 +112,23 @@ typedef enum {
...
+    cc2538_ioc_sel_t txd;     /**< function assigned to the TX pin */
+    gpio_t cts_pin;           /**< CTS pin - set to GPIO_UNDEF when not using HW flow control */
+    gpio_t rts_pin;           /**< RTS pin - set to GPIO_UNDEF when not using HW flow control */
```
Comment (haukepetersen) [inline cpu/cc2538/include/periph_cpu.h]: These lines are exceeding the 80 character line length...

File: cpu/cc2538/periph/uart.c [hunk 13]
```diff
@@ -61,26 +61,23 @@ enum {
 #define DIVFRAC_MASK          ( (1 << DIVFRAC_NUM_BITS) - 1 )
 
 /** @brief Indicates if there are bytes available in the UART0 receive FIFO */
-#define uart0_rx_avail() ( UART0->cc2538_uart_fr.FRbits.RXFE == 0 )
+#define uart0_rx_avail() ( uart_config[0].dev->cc2538_uart_fr.FRbits.RXFE == 0 )
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: style: either use capitalized define names here, or use static inline functions...

BUT: these 4 functions are not used in the code, so drop them!

File: cpu/cc2538/periph/uart.c [hunk 14]
```diff
@@ -61,26 +61,23 @@ enum {
...
 /**
  * @brief Allocate memory to store the callback functions.
  */
-static uart_isr_ctx_t uart_config[UART_NUMOF];
-
-cc2538_uart_t * const UART0 = (cc2538_uart_t *)0x4000c000;
-cc2538_uart_t * const UART1 = (cc2538_uart_t *)0x4000d000;
+static uart_isr_ctx_t uart_ctx[UART_NUMOF];
 
 /*---------------------------------------------------------------------------*/
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: I would prefer to drop those 'separators'. We never really use anything like it in the RIOT code base, so they seem to be out of another world :-)

File: cpu/cc2538/periph/uart.c [hunk 15]
```diff
@@ -106,49 +103,6 @@ static void reset(cc2538_uart_t *u)
...
-    if (mis & (OEMIS | BEMIS | FEMIS)) {
-        /* ISR triggered due to some error condition */
-        reset(UART_1_DEV);
-    }
-
-    cortexm_isr_end();
-}
-#endif /* UART_1_EN */
 
 static int init_base(uart_t uart, uint32_t baudrate);
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: please dropt the `init_base()` function and integrate it directly into `uart_init()`. This is a leftover from some former refactoring and should be resolved.

File: cpu/cc2538/periph/uart.c [hunk 16]
```diff
@@ -161,26 +115,12 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
...
-#if UART_1_EN
-        case UART_1:
-            NVIC_SetPriority(UART1_IRQn, UART_IRQ_PRIO);
-            NVIC_EnableIRQ(UART1_IRQn);
-            break;
-#endif
-        default:
-            return UART_NODEV;
-    }
+    NVIC_SetPriority(uart_config[uart].irq, UART_IRQ_PRIO);
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: drop this line. No need for setting the priority, as this is done globally for all interrupts in the cortexm initialization. (also some ancient leftover...)

File: cpu/cc2538/periph/uart.c [hunk 17]
```diff
@@ -189,30 +129,12 @@ static int init_base(uart_t uart, uint32_t baudrate)
 {
     cc2538_uart_t *u = NULL;
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: no need to first assign `NULL` and then assign a different value in the line below... Just write `cc2538_uart_t *u = uart_config[uart].dev;` directly in one line.

Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: But also here and in all other UART functions: please add assertions for validating the given parameter(s), as in
```c
assert(uart < UART_NUMOF);
```

File: cpu/cc2538/periph/uart.c [hunk 18]
```diff
@@ -189,30 +129,12 @@ static int init_base(uart_t uart, uint32_t baudrate)
...
-            break;
-#endif
-
-        default:
-            (void)u;
-            return UART_NODEV;
-    }
+    u = uart_config[uart].dev;
+    gpio_init_af(uart_config[uart].rx_pin, uart_config[uart].rxd, GPIO_IN);
+    gpio_init_af(uart_config[uart].tx_pin, uart_config[uart].txd, GPIO_OUT);
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: as stated above, the values for `rxd` and `txd` are static as far as I can see, so no need to have them in the board config.

File: cpu/cc2538/periph/uart.c [hunk 19]
```diff
@@ -189,30 +129,12 @@ static int init_base(uart_t uart, uint32_t baudrate)
...
 
-#if UART_0_EN || UART_1_EN
     /* Enable clock for the UART while Running, in Sleep and Deep Sleep */
-    unsigned int uart_num = ( (uintptr_t)u - (uintptr_t)UART0 ) / 0x1000;
+    unsigned int uart_num = ( (uintptr_t)u - (uintptr_t)uart_config[uart].dev ) / 0x1000;
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: `s/uart_config[uart].dev/u/`

Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: wait, ´u` is assigned with `uart_config[uart].dev` a few lines above, so this line seems to be evaluating always to `0`, right? This seems to be wrong.

Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: Ok, now I see: you have to leave this line unchanged, as it is computing the UART number from their base register offsets, and in this case `UART0` is needed here.

File: cpu/cc2538/periph/uart.c [hunk 20]
```diff
@@ -224,20 +146,18 @@ static int init_base(uart_t uart, uint32_t baudrate)
...
         u->cc2538_uart_ctl.CTLbits.CTSEN = 1;
-#endif
     }
 
     /* Enable clock for the UART while Running, in Sleep and Deep Sleep */
-    uart_num = ( (uintptr_t)u - (uintptr_t)UART0 ) / 0x1000;
+    uart_num = ( (uintptr_t)u - (uintptr_t)uart_config[0].dev ) / 0x1000;
     SYS_CTRL_RCGCUART |= (1 << uart_num);
     SYS_CTRL_SCGCUART |= (1 << uart_num);
     SYS_CTRL_DCGCUART |= (1 << uart_num);
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: this code block is duplicated (see above). This needs to be cleaned up.

File: cpu/cc2538/periph/uart.c [hunk 21]
```diff
@@ -316,3 +222,37 @@ void uart_poweroff(uart_t uart)
 {
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: you should also implement the power_on and power_off functions while you at it. Probably they can be just on-liners as in 
````c
assert(uart < UART_NUMOF);
uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = [0|1];
```

Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: Also, bitmasks rather than each bit on one line

**Comment by danpetry** [comment]
Hello all, ready for review prior to squashing

File: boards/cc2538dk/include/periph_conf.h (hunk 9 above)
Comment (aabadie) [reply boards/cc2538dk/include/periph_conf.h]: Well I'm with @haukepetersen, this comment should be removed or updated (e.g USB port or something more hardware related).

Comment (danpetry) [reply boards/cc2538dk/include/periph_conf.h]: Sorry, have removed that now!

File: cpu/cc2538/periph/uart.c (hunk 21 above)
Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: ok, I've done that. Also the clock on/off. However, they're untested...

File: cpu/cc2538/periph/uart.c [hunk 22]
```diff
@@ -276,28 +160,21 @@ static int init_base(uart_t uart, uint32_t baudrate)
     /* UART Enable */
     u->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 
+    /* register callbacks */
+    uart_ctx[uart].rx_cb = rx_cb;
+    uart_ctx[uart].arg = arg;
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: This is very dangerous: setting the callbacks after enabling the UART device can lead unwanted behavior: if the driver is initialized a second time (e.g. to change the baudrate), the global NVIC interrupt is already enabled. So we could get a call to a wrong isr context.

Solution a) just move the two lines above before the UARTEN
Solution b) disable the gloabl interrupt source at the beginning of the initialization

File: cpu/cc2538/periph/uart.c [hunk 23]
```diff
@@ -276,28 +160,21 @@ static int init_base(uart_t uart, uint32_t baudrate)
...
+
+    /* enable RX interrupt */
+    NVIC_EnableIRQ(UART_IRQ(uart_num));
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: Just noticed something else: the RX part of the UART driver **must** only be initialized, if the callback parameter is not `NULL`. See the documentation in `drivers/periph/uart.h`:
>  If no callback parameter is given (rx_cb := NULL), the UART will be
>  initialized in TX only mode

So we need to factor out all the RX related code in something like
```c
if (rx_cb) {
    uart_ctx[uart].rx_cb = rx_cb;
    ...
    RXEN = 1
    ...
    NVIC_Enable ...
```

Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: Unrelated to this PR: but we should also consider how we include this in our testing...

File: cpu/cc2538/periph/uart.c (hunk 21 above)
Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: test them :-)

One thing that just came to mind: it might be useful to make sure that the TX FIFO is empty before putting the device into poweroff mode, right?

File: cpu/cc2538/periph/uart.c [hunk 24]
```diff
@@ -308,11 +185,64 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
...
+    mis = uart_config[uart].dev->cc2538_uart_mis.MIS;
+    uart_config[uart].dev->ICR = mis;
+
+    while (uart_config[uart].dev->cc2538_uart_fr.FRbits.RXFE == 0) {
+        uart_ctx[uart].rx_cb(uart_ctx[uart].arg, uart_config[uart].dev->DR);
+    }
+
+    if (mis & (OEMIS | BEMIS | FEMIS)) {
+        /* ISR triggered due to some error condition */
+        reset(uart_config[uart].dev);
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: I stumbled upon this in my last review, though I stayed quite. It seems to me, that the call to this reset function is broken in some regards:
- reset always enables RX mode, independent if it was enabled before (see my comment above)
- so we really need to clear the FIFOs on error? TX FIFO overrun is not possible in any case, and RX overrun is very unlikely, and even if it happens, I don't think we should throw away the already received data...

My approach here would be to remove the `reset()` function, and simply reset error bits

File: cpu/cc2538/periph/uart.c (hunk 21 above)
Comment (haukepetersen) [reply cpu/cc2538/periph/uart.c]: #8447 makes testing this hopefully easier for you...

File: cpu/cc2538/periph/uart.c (hunk 22 above)
Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: I went for a). Less instructions and therefore code size...

File: cpu/cc2538/periph/uart.c (hunk 23 above)
Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: Ok. Rather than put if (rx_cb) all over the place, I've only made this control whether we enable/disable the Rx interrupt. Same reason as above. This would satisfy the statement in drivers/periph/uart.h. Was also thinking about whether to remove the rx pin configuration in case there was a conflict but reckoned that this would be up to the user of the driver to resolve...

In terms of testing, rx_cb is always defined in the current tests/periph_uart... we could provide another argument to cmd_init which initialises the driver with or without the callback function defined...?

Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: NB that in the commits I've cherry picked your updated uart test application. Also, I've added clock on/off tests to them. I needed to change include/uart.h though for this...

File: cpu/cc2538/periph/uart.c (hunk 24 above)
Comment (danpetry) [reply cpu/cc2538/periph/uart.c]: Hmm yeah. So you are proposing just to discard the data byte and not re-initialise the hardware. I found [this discussion](https://electronics.stackexchange.com/questions/29134/handling-of-uart-errors) which basically says that you could pass these errors up to the higher level if we really want to ensure data integrity, but these errors are rare. Is there any case in which these errors can be generated by faulty hardware on this side? I don't know of any ways so my fixup just resets the error bits.

File: cpu/cc2538/periph/uart.c [hunk 25]
```diff
@@ -308,11 +173,72 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
...
-    (void) uart;
+    assert(uart < UART_NUMOF);
+
+    /* Wait for the TX FIFO to clear */
+    while (uart_config[uart].dev->cc2538_uart_fr.FRbits.TXFF) {}
+
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 0;
 }
+
+void uart_clockon(uart_t uart)
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: What are these two functions (clockon/clockoff) about? They are not part of the UART interface, so they should not be non-static. Further they are not used anywhere in the code, so why have them? I suspect that the code from this functions should probalby rather be integrated into the poweron/off functions, right?!

**Comment by haukepetersen** [comment]
re-triggered Murdock

File: cpu/cc2538/periph/uart.c [hunk 26]
```diff
@@ -46,173 +62,33 @@ enum {
...
-static int init_base(uart_t uart, uint32_t baudrate);
+static uart_isr_ctx_t uart_ctx[UART_NUMOF];
 
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
-    /* initialize basic functionality */
-    int res = init_base(uart, baudrate);
-    if (res != UART_OK) {
-        return res;
-    }
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: remove this newline

File: cpu/cc2538/periph/uart.c [hunk 27]
```diff
@@ -308,11 +173,61 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
...
+    SYS_CTRL->cc2538_sys_ctrl_unnamed1.RCGCUART &= ~(1 << uart);
 }
+
+static inline void irq_handler(uart_t uart)
+{
+    assert(uart < UART_NUMOF);
+
+    cc2538_uart_t *u = uart_config[uart].dev;
+
+    uint_fast16_t mis;
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: any special reason for using the (in the RIOT context very uncommon) `uint_fast16_t` instead of simply `uint16_t` here?

File: cpu/cc2538/periph/uart.c [hunk 28]
```diff
@@ -308,11 +173,61 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
...
+
+    /* Latch the Masked Interrupt Status and clear any active flags */
+    mis = uart_config[uart].dev->cc2538_uart_mis.MIS;
```
Comment (haukepetersen) [inline cpu/cc2538/periph/uart.c]: maybe to save two lines above, how about doing
```c
uint16_t mis = uart_config[uart].dev->cc2538_uart_mis.MIS;
```
here?

**Comment by danpetry** [comment]
Changed them, rebased and squashed

**Comment by danpetry** [comment]
There are some instructions on testing a few comments up ^^^^

**Comment by kYc0o** [comment]
Tested on cc2538 with success. Is it ready for merging?

**Comment by danpetry** [comment]
W00t first PR!

I would like to ask though, if anyone has a remote to test on, for the reasons given in https://github.com/RIOT-OS/RIOT/pull/8345#issuecomment-357687760 ?

Or, if not, is there a way of recording that this hasn't been tested?

**Comment by danpetry** [comment]
Hello @smlng  and @PeterKietzmann . Would either of you be able to test the remotes? Specifically the test to do with setting the receive callback function to NULL? This hasn't been tested as I don't have a board, so if someone tries to use it in a few months/years/whatever and it doesn't work properly, they'll have to spend a bunch of time debugging it.

**Comment by smlng** [comment]
I'll see to it and tests remote revb, today.

**Comment by danpetry** [comment]
Awesome thank you

**Comment by smlng** [comment]
reACK from my side, just tested with remote-revb w/ and w/o rx_cb == NULL, results are as expected. As the remotes share their periph.conf and reva and pa are outdated, I would say this suffices to get this merged. However, I'll (re)trigger Murdock.

**Comment by kYc0o** [comment]
Congrats @danpetry for your first contribution!

**Comment by danpetry** [comment]
@smlng thanks for testing. Yes I agree this should be fine.
